### PR TITLE
James-3385: update Email/Query documentation

### PR DESCRIPTION
**Pull request by nvduc91**
I still have some confused, please help me.

### CODE REVIEW

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 1]
```diff
@@ -639,15 +661,27 @@ and response:
 
 ## Email/changes
 
+<aside class="warning">
+    We not implemented yet
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:665]: s/We not implemented yet/Not implemented yet/ everywhere

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 2]
```diff
@@ -639,15 +661,27 @@ and response:
...
+    Partially implemented
+</aside>
+
 This is a standard "/query" method as described in [@!RFC8620], Section 5.5, but with the following additional request arguments:
 
 - **collapseThreads**: `Boolean` (default: false)
   If `true`, Emails in the same Thread as a previous Email in the list (given the filter and sort order) will be removed from the list. This means only one Email at most will be included in the list for any given Thread.
 
+<aside class="warning">
+   not implemented yet
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:682]: Each message belonging to a single thread according to our current implementation, collapseThread is a noop operation.

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 3]
```diff
@@ -697,6 +731,21 @@ A **FilterCondition** object has the following properties, any of which may be o
 
 If zero properties are specified on the FilterCondition, the condition MUST always evaluate to `true`. If multiple properties are specified, ALL must apply for the condition to be `true` (it is equivalent to splitting the object into one-property conditions and making them all the child of an AND filter operator).
 
+<aside class="warning">
+Following those properties are not supported yet:
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:735]: Those properties are not supported yet for filtering:

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 4]
```diff
@@ -728,6 +777,17 @@ The following values for the *property* field on the Comparator object SHOULD be
 - **allInThreadHaveKeyword** - This value MUST be considered `true` for the Email if **all** of the Emails in the same Thread have the keyword given as an additional *keyword* property on the *Comparator* object.
 - **someInThreadHaveKeyword** - This value MUST be considered `true` for the Email if **any** of the Emails in the same Thread have the keyword given as an additional *keyword* property on the *Comparator* object.
 
+<aside class="warning">
+ Sort by following properties are not supported yet:
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:781]: Those properties are not supported yet for sorting:

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 5]
```diff
@@ -750,15 +810,27 @@ This would sort Emails in flagged Threads first (the Thread is considered flagge
 
 When *collapseThreads* is `true`, then after filtering and sorting the Email list, the list is further winnowed by removing any Emails for a Thread id that has already been seen (when passing through the list sequentially). A Thread will therefore only appear **once** in the result, at the position of the first Email in the list that belongs to the Thread (given the current sort/filter).
 
+<aside class="warning">
+We are not supported yet
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:814]: Body truncation implementation do not take html structure into account.

Truncation can happen in the middle of an html tag.

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 6]
```diff
@@ -363,7 +363,7 @@ An **EmailBodyPart** object has the following properties:
 In addition, the client may request/send EmailBodyPart properties representing individual header fields, following the same syntax and semantics as for the Email object, e.g., `header:Content-Type`.
 
 <aside class="warning">
-// I have some confused: this part we already supported
+ We support it for Email, but not yet for EmailBodyPart
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:366]: We support fetching specific headers for the Email entity, but this is not yet implemented for the EmailBodyPart entity.

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by nvduc91** [comment]
rebase master

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 7]
```diff
@@ -697,6 +732,21 @@ A **FilterCondition** object has the following properties, any of which may be o
 
 If zero properties are specified on the FilterCondition, the condition MUST always evaluate to `true`. If multiple properties are specified, ALL must apply for the condition to be `true` (it is equivalent to splitting the object into one-property conditions and making them all the child of an AND filter operator).
 
+<aside class="warning">
+Those properties are not supported yet for filtering:
```
Comment (LanKhuat) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:736]: Should be "These"

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 8]
```diff
@@ -728,6 +778,17 @@ The following values for the *property* field on the Comparator object SHOULD be
 - **allInThreadHaveKeyword** - This value MUST be considered `true` for the Email if **all** of the Emails in the same Thread have the keyword given as an additional *keyword* property on the *Comparator* object.
 - **someInThreadHaveKeyword** - This value MUST be considered `true` for the Email if **any** of the Emails in the same Thread have the keyword given as an additional *keyword* property on the *Comparator* object.
 
+<aside class="warning">
+ Those properties are not supported yet for sorting:
```
Comment (LanKhuat) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:782]: Idem

**Review by LanKhuat — APPROVED** [review]

File: server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown [hunk 9]
```diff
@@ -639,15 +662,27 @@ and response:
...
+
 This is a standard "/query" method as described in [@!RFC8620], Section 5.5, but with the following additional request arguments:
 
 - **collapseThreads**: `Boolean` (default: false)
   If `true`, Emails in the same Thread as a previous Email in the list (given the filter and sort order) will be removed from the list. This means only one Email at most will be included in the list for any given Thread.
 
+<aside class="warning">
+  Each message belonging to a single thread according to our current implementation, collapseThread is a noop operation.
+</aside>
+
```
Comment (remk) [inline server/protocols/jmap-rfc-8621/doc/specs/spec/mail/message.mdown:685]: we can add that the calcultateTotal is not supported and so nor are the negative position

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Review by remk — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**